Messaging.md

```diff
@@ -165,10 +173,69 @@ This type contains signatures of the funding transaction and any necessary infor
 `witness` is the data for a witness element in a witness stack. An empty `witness_stack` is an error,
 as every input must be Segwit. Witness elements should *not* include their length as part of the witness data.
 
+### The `event_descriptor` Type
+
+This type contains information about the outcomes in an event for which an oracle plans on releasing a signature over.
+
+#### Version 0 `external_event_descriptor`
+
+1. type: 55300 (`external_event_descriptor_v0`)
+2. data:
+   * [`utf8_string`:`string`]
+
+`string` can refer to anything here and it is up to the oracle and user to agree on how to interpret it.
+
+#### Version 0 `enum_event_descriptor`
+
+1. type: 55302 (`enum_event_descriptor_v0`)
+2. data:
+   * [`u16`:`outcome_1_len`]
+   * [`utf8_string`:`outcome_1`]
+   * ...
+   * [`u16`:`outcome_n_len`]
+   * [`utf8_string`:`outcome_n`]
+
+This type of event descriptor is a simple enumeration where the value `n` is omitted from being explicitly included as it can be derived from the length field of the TLV.
```

Comment:
Actually on this occasion `n`  should probably be included because length can't be derived extrinsically due to variable internal length. (It was included in funding_signatures above, for example)

Reply:
We still shouldn't need `n` because of the `Length` in the TLV. However I will clarify that in this line

Reply:
I don't understand how length in the TLV is helpful in determining `n`?

Reply:
and re your clarification you cannot just parse the individual lengths because that requires going through the whole blob (you don't know where the lengths are going to be a priori)

Reply:
This is my implementation of it, this should be doable for everyone:

```
    val iter = ValueIterator(value)
    val builder = Vector.newBuilder[String]

    while (iter.index < value.length) {
      val len = UInt16(iter.takeBits(16))
      val outcomeBytes = iter.take(len.toInt)
      val str = new String(outcomeBytes.toArray, StandardCharsets.UTF_8)
      builder.+=(str)
    }

    EnumEventDescriptorV0TLV(builder.result())
```

Reply:
I would like to check that this is feasible with ldk as is, please remind me if I don't update.

Reply:
I agree that it isn't absolutely vital to include `n` but I think it is good practice to allow length to be known without requiring linear time computation, the trade-off of adding a UInt16 to the front seems worth it and is good practice

Reply:
I guess it's just 2 bytes can just add it.

Reply:
Added.